File: database/database-api-model/src/main/java/io/ehdev/conrad/database/model/user/ApiUserPermission.java
package io.ehdev.conrad.database.model.user;

import org.jetbrains.annotations.NotNull;


public enum ApiUserPermission {
    NONE(0),
    READ(1 << (0)),
    WRITE(1 << (1)),
    ADMIN(1 << (2));

    int securityId;

    ApiUserPermission(int id) {
        this.securityId = id;
    }

    public int getSecurityId() {
        return securityId;
    }

    @NotNull public static ApiUserPermission findById(int id) {
        for (ApiUserPermission value : values()) {
            if (value.securityId == id) {
                return value;
            }
        }
        throw new RuntimeException(id + " is not a known permission level");
    }
}

File: database/database-api-model/src/main/kotlin/io/ehdev/conrad/database/model/user/ApiUserPermission.kt
package io.ehdev.conrad.database.model.user

enum class ApiUserPermission {
    NONE(0),
    READ(1.shl(0)),
    WRITE(1.shl(1)),
    ADMIN(1.shl(2));

    val securityId: Int

    constructor(id: Int) {
        this.securityId = id
    }

    companion object {
        @JvmStatic fun findById(id: Int): ApiUserPermission? {
            for (value in values()) {
                if(value.securityId == id) {
                    return value
                }
            }
            return null;
        }
    }
}
